tsfile/src/main/java/org/apache/iotdb/tsfile/read/query/dataset/QueryDataSet.java

```diff
@@ -102,7 +102,9 @@ protected void initQueryDataSetFields(
     this.columnNum = 0;
     if (paths != null) {
       for (Path p : paths) {
-        columnNum += p.getColumnNum();
+        if (p != null) {
```

Comment:
```suggestion
        if (p != null) {
```
Please add some comments here.